### PR TITLE
cumulate backports from master

### PR DESCRIPTION
**Pull request by marcusmueller**

### CODE REVIEW

File: grc/core/blocks/embedded_python.py [hunk 1]
```diff
@@ -149,10 +149,10 @@ def rewrite(self):
     def _update_params(self, params_in_src):
         param_factory = self.parent_platform.make_param
         params = {}
-        for param in list(self.params):
-            if hasattr(param, '__epy_param__'):
-                params[param.key] = param
-                del self.params[param.key]
+        for key, value in self.params.copy().items():
```
Comment (mbr0wn) [inline grc/core/blocks/embedded_python.py]: Do we know if this works with Python 2?

Comment (marcusmueller) [reply grc/core/blocks/embedded_python.py]: true, I should have tested the thing under Py2k; don't even have a system at hand to do that, and no test coverage of that, so can't do testing on CI...

**Comment by marcusmueller** [comment]
I'll check whether emb Py blocks work on Py2 in a VM.

**Comment by devnulling** [comment]
Need to add https://github.com/gnuradio/gnuradio/pull/2927

**Comment by devnulling** [comment]
Need to add https://github.com/gnuradio/gnuradio/pull/2821

**Comment by devnulling** [comment]
Need to add: https://github.com/gnuradio/gnuradio/pull/2913  https://github.com/gnuradio/gnuradio/issues/2962

**Comment by mbr0wn** [comment]
Let's open a new PR for those.

**Comment by mbr0wn** [comment]
Manually merged the clear commits. Will take the embedded python commit along with others that @devnulling mentioned and create a new backport PR.

**Comment by mbr0wn** [comment]
@devnulling all the commits you mentioned coming up...

**Pull request closed without merging**